xarray/namedarray/core.py

```diff
@@ -348,6 +350,30 @@ def _dask_finalize(
         data = array_func(results, *args, **kwargs)
         return type(self)(self._dims, data, attrs=self._attrs)
 
+    def get_axis_num(self, dim: Hashable | Iterable[Hashable]) -> int | tuple[int, ...]:
+        """Return axis number(s) corresponding to dimension(s) in this array.
+
+        Parameters
+        ----------
+        dim : str or iterable of str
+            Dimension name(s) for which to lookup axes.
+
+        Returns
+        -------
+        int or tuple of int
+            Axis number or numbers corresponding to the given dimensions.
+        """
+        if not isinstance(dim, str) and isinstance(dim, Iterable):
+            return tuple(self._get_axis_num(d) for d in dim)
+        else:
+            return self._get_axis_num(dim)
+
+    def _get_axis_num(self: Any, dim: Hashable) -> int:
+        try:
+            return self.dims.index(dim)
+        except ValueError:
+            raise ValueError(f"{dim!r} not found in array dimensions {self.dims!r}")
```

Comment:
explicitly raise from previous error?

```suggestion
        except ValueError as exc:
            raise ValueError(f"{dim!r} not found in array dimensions {self.dims!r}") from exc
```

Reply:
I suspect this is intentional to generate a smaller traceback.